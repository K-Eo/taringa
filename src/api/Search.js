import _ from 'lodash'

import APIBase from './APIBase'

class Search extends APIBase {
  story(q, cancelToken, params = {}) {
    return super.execute(
      '/search/story',
      _.assign({}, params, { q }),
      cancelToken
    )
  }

  user(q, cancelToken, params = {}) {
    return super.execute(
      '/search/user',
      _.assign({}, params, { q }),
      cancelToken
    )
  }

  channel(q, cancelToken, params = {}) {
    return super.execute(
      '/search/channel',
      _.assign({}, params, { q }),
      cancelToken
    )
  }
}

export default Search
