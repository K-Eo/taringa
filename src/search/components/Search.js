import React, { Component } from 'react'

import Ad from '../../ads/components/Ad'
import TabButton from './TabButtonContainer'

const tabs = {}

class Search extends Component {
  constructor(props) {
    super(props)

    this.state = {
      activeTab: 'stories',
    }

    this.handleTabClick = this.handleTabClick.bind(this)
  }

  handleTabClick(e) {
    this.setState({ activeTab: e })
  }

  render() {
    const { activeTab } = this.state

    const CurrentTab = tabs[activeTab]

    return (
      <div className="row">
        <div className="col-12 col-lg-8">
          <div className="card">
            <div className="card-header">
              <div className="nav nav-tabs card-header-tabs" role="tablist">
                <TabButton
                  id="stories"
                  isActive={activeTab === 'stories'}
                  label="Posts"
                  onClick={this.handleTabClick}
                />

                <TabButton
                  id="users"
                  isActive={activeTab === 'users'}
                  label="Usuarios"
                  onClick={this.handleTabClick}
                />

                <TabButton
                  id="channels"
                  isActive={activeTab === 'channels'}
                  label="Canales"
                  onClick={this.handleTabClick}
                />
              </div>
            </div>

            <div className="card-body">{CurrentTab && <CurrentTab />}</div>
          </div>
        </div>

        <div className="col-4 d-none d-lg-block">
          <Ad />
        </div>
      </div>
    )
  }
}

export default Search
