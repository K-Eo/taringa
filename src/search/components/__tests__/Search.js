import React from 'react'

import Search from '../Search'

describe('Search', () => {
  it('exists', () => {
    expect(Search).toBeDefined()
  })

  it('renders', () => {
    const tree = renderer.create(<Search />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
