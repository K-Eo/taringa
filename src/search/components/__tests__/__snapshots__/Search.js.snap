// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="nav nav-tabs card-header-tabs"
          role="tablist"
        >
          <Connect(TabButton)
            id="stories"
            isActive={true}
            label="Posts"
            onClick={[Function]}
          />
          <Connect(TabButton)
            id="users"
            isActive={false}
            label="Usuarios"
            onClick={[Function]}
          />
          <Connect(TabButton)
            id="channels"
            isActive={false}
            label="Canales"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="card-body"
      />
    </div>
  </div>
  <div
    className="col-4 d-none d-lg-block"
  >
    <Ad />
  </div>
</div>
`;
